File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/playback/Item.java
/**
 * @file
 * @brief Interface for playback item (metadata only)
 * @author vitamin.caig@gmail.com
 */

package app.zxtune.playback;

import android.net.Uri;

import app.zxtune.core.Identifier;
import app.zxtune.TimeStamp;

public interface Item {

  /**
   * @return Unique item identifier - may be equal to dataId in case of plain file playback
   */
  Uri getId();

  /**
   * @return Item's data identifier
   */
  Identifier getDataId();

  /**
   * @return Item's title (possibly empty)
   */
  String getTitle();

  /**
   * @return Item's author (possibly empty)
   */
  String getAuthor();

  /**
   * @return Item's program information (possibly empty)
   */
  String getProgram();

  /**
   * @return Item's comment (possibly empty)
   */
  String getComment();

  /**
   * @return Item's internal strings (possibly empty)
   */
  String getStrings();

  /**
   * @return Item's duration
   */
  TimeStamp getDuration() throws Exception;
}

File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/playback/Item.kt
/**
 * @file
 * @brief Interface for playback item (metadata only)
 * @author vitamin.caig@gmail.com
 */
package app.zxtune.playback

import android.net.Uri
import app.zxtune.TimeStamp
import app.zxtune.core.Identifier

interface Item {
    /**
     * @return Unique item identifier - may be equal to dataId in case of plain file playback
     */
    val id: Uri

    /**
     * @return Item's data identifier
     */
    val dataId: Identifier

    /**
     * @return Item's title (possibly empty)
     */
    val title: String

    /**
     * @return Item's author (possibly empty)
     */
    val author: String

    /**
     * @return Item's program information (possibly empty)
     */
    val program: String

    /**
     * @return Item's comment (possibly empty)
     */
    val comment: String

    /**
     * @return Item's internal strings (possibly empty)
     */
    val strings: String

    /**
     * @return Item's duration
     */
    val duration: TimeStamp
}
